### PR TITLE
Copy data processings

### LINKED ISSUE #2: Copy objects from Data Processing module
#### Problem statement
Currently, the toolkit does not allow to copy objects from the Data Processing module to another workspace. 

#### Proposed solution
The idea is to follow the same pattern as the other modules, and enable copying objects in the same way.

### PR DESCRIPTION
**Pull request by drscholly**
Fixes #2 

#### Checklist

- [x] I have read the [Contribution Guide](../blob/main/CONTRIBUTING.md) ;
- [x] I have read the [Contributor License Agreement (CLA)](../blob/main/CONTRIBUTOR_LICENSE_AGREEMENT.md) and understand that the submission of this PR will constitute an  electronic signature of my agreement of the terms and conditions of DataGalaxy's CLA ;
- [x] There is an approved issue describing the change when contributing a new feature ;
- [x] I have added tests that prove my fix is effective or that my feature works ;
- [x] I have added necessary documentation (if appropriate) ;

#### Short description of what this resolves:
The toolbox now allows to copy all dataProcessing objects from a workspace to another. It includes the dataProcessingItems, but it requires that the dictionary was previously copied as the links will not be created otherwise. 

#### Changes proposed in this pull request:

- New API methods to fetch and copy dataProcessings
- New parser option to copy dataProcessings
- A simple test

### CODE REVIEW

File: toolbox/api/datagalaxy_api_dataprocessings.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+class DataGalaxyApiDataprocessings:
+    def __init__(self, url: str, access_token: str, workspace: dict):
+        self.url = url
+        self.access_token = access_token
+        self.workspace = workspace
+
+    def list_dataprocessings(self, workspace_name: str) -> list:
+        if self.workspace["isVersioningEnabled"]:
+            raise Exception('Workspace with versioning enabled are currently not supported.')
```
Comment (ggrossetie) [inline toolbox/api/datagalaxy_api_dataprocessings.py:15]: move this logic in the constructor

File: toolbox/commands/copy_dataprocessings.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        access_token=source_access_token,
+        workspace=workspaces_api_on_source_env.get_workspace(workspace_source_name)
+    )
+    dataprocessings_on_target_workspace = DataGalaxyApiDataprocessings(
+        url=url_target,
+        access_token=target_access_token,
+        workspace=workspaces_api_on_target_env.get_workspace(workspace_target_name)
+    )
+
+    if workspaces_api_on_target_env.get_workspace(workspace_target_name):
```
Comment (ggrossetie) [inline toolbox/commands/copy_dataprocessings.py:43]: introduce a variable and check that both the source and target workspace exist

File: toolbox/commands/copy_dataprocessings.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        # fetching dataprocessings from source workspace
+        workspace_source_dataprocessings = dataprocessings_on_source_workspace.list_dataprocessings(workspace_source_name)
+
+        for dp in workspace_source_dataprocessings:
+            dp_index = workspace_source_dataprocessings.index(dp)
+            items = dataprocessings_on_source_workspace.list_dataprocessing_items(workspace_name=workspace_source_name, parent_id=dp['id'])
+            for item in items:
```
[the remark this answers is not in the document]
Comment (drscholly) [reply toolbox/commands/copy_dataprocessings.py]: There is already a logger in the `list_dataprocessing_items` method, do you believe it is necessary to add another logger here ?

**Review by ggrossetie — APPROVED** [review]
LGTM

**Pull request merged**